LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaPayloadDecoder.cs

```diff
@@ -56,7 +56,19 @@ public async ValueTask<DecodePayloadResult> DecodeMessageAsync(string devEUI, by
             var base64Payload = ((payload?.Length ?? 0) == 0) ? string.Empty : Convert.ToBase64String(payload);
 
             // Call local decoder (no "http://" in SensorDecoder)
-            if (!sensorDecoder.Contains("http://", StringComparison.Ordinal))
+            if (Uri.TryCreate(sensorDecoder, UriKind.Absolute, out var url) && (url.Scheme is "http" or "https"))
+            {
+                // Support decoders that have a parameter in the URL
+                // http://decoder/api/sampleDecoder?x=1 -> should become http://decoder/api/sampleDecoder?x=1&devEUI=11&fport=1&payload=12345
+
+                var query = HttpUtility.ParseQueryString(url.Query);
+                query["devEUI"] = devEUI;
+                query["fport"] = fport.ToString(CultureInfo.InvariantCulture);
+                query["payload"] = base64Payload;
```

Comment:
I was checking if the `Query` class does the same encoding by default as this was the previous implementation.  According to [the docs](https://docs.microsoft.com/en-us/dotnet/api/system.uribuilder.query?view=net-6.0) they do RFC2396 - not sure if this the case also for HttpUtility.UrlEncode? Checking for example `HttpUtility.UrlEncode("{hello:world}");` vs `query["foo"] = "{hello:world}";` yields different results. 

Not sure if this could be a problem for the payload (not sure what characters are sent). Do you think we have sufficient test coverage here to trust the tests? Should we run an E2E test for this?

Reply:
Given this test:

```c#
var url = new Uri("http://example.com/");
var query = HttpUtility.ParseQueryString(url.Query);
const string foo = "{hello:world}";
query["foo"] = foo;
Console.WriteLine(query.ToString());
Console.WriteLine(HttpUtility.UrlEncode(foo));
```

I see no difference in how the _value_ of `foo` is encoded in either case. Output of [running the above](https://dotnetfiddle.net/gEQ3EI):

    foo=%7bhello%3aworld%7d
    %7bhello%3aworld%7d

Reply:
This is what I was checking
```
var a = "http://hello/world?x=1/";
Uri.TryCreate(a, UriKind.Absolute, out var url);
var query = HttpUtility.ParseQueryString(url.Query);
const string foo = "{hello:world}";
query["foo"] = foo;
new UriBuilder(url) { Query = query.ToString() }.Uri.Dump();

HttpUtility.UrlEncode(foo).Dump();
```

Reply:
As discussed with Atif, our understanding is that `query.ToString()` over-escapes and that {} are not reserved characters anyway. Since we are passing a "strong" object Uri and also by trying [the rest of reserved characters](https://datatracker.ietf.org/doc/html/rfc2396#section-3.4), we think this usage is ok.

Reply:
> Regarding the trailing slash, we discussed that it would be better to remove here as well in order to be backwards compatible.

Reverted with 2d222b0d0a88142b33dd298f27c787bd47cc88f9.